[expo-barcode-scanner] ArrayIndexOutOfBoundsException

## 🐛 Bug Report

### Environment

<!-- In your terminal run `expo diagnostics` and paste the output here. -->
```
Expo CLI 3.13.1 environment info:
    System:
      OS: macOS 10.15.3
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 10.16.1 - ~/.nvm/versions/node/v10.16.1/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.9.0 - ~/.nvm/versions/node/v10.16.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6241897
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      @sentry/react-native: 1.3.3 => 1.3.3 
      react: 16.12.0 => 16.12.0 
      react-native: 0.61.5 => 0.61.5 
      react-navigation: ^4.2.2 => 4.2.2 
    npmGlobalPackages:
      expo-cli: 3.13.1
```
<!-- Please also let us know about your app's target (iOS, Android, Client, Standalone, ExpoKit) -->

### Steps to Reproduce
I cannot reproduce, but there's many of my app users getting this error
![Screen Shot 2020-04-25 at 19 06 57](https://user-images.githubusercontent.com/6567775/80279445-f3ae9280-8727-11ea-8104-e0f0b6e04179.png)
![Screen Shot 2020-04-25 at 19 06 53](https://user-images.githubusercontent.com/6567775/80279457-01641800-8728-11ea-8f0a-5c9d235264a8.png)


<!--
  How would you describe your issue to someone who doesn’t know you or your project?
  Try to write a sequence of steps that anybody can repeat to see the issue.
  Be specific! If the bug cannot be reproduced, your issue may be closed.
-->

### Expected Behavior
```
Caused by java.lang.ArrayIndexOutOfBoundsException
length=55296; index=56159
expo.modules.barcodescanner.scanners.ZxingBarCodeScanner.scan (ZxingBarCodeScanner.java:57)
expo.modules.camera.tasks.BarCodeScannerAsyncTask.doInBackground (BarCodeScannerAsyncTask.java:36)
expo.modules.camera.tasks.BarCodeScannerAsyncTask.doInBackground (BarCodeScannerAsyncTask.java:6)
android.os.AsyncTask$3.call (AsyncTask.java:378)
```
<!--
  How did you expect your project to behave?
  It’s fine if you’re not sure your understanding is correct.
  Just write down what you thought would happen.
-->

### Actual Behavior
No crash

I think is it possible I make a pull request like this?
in https://github.com/expo/expo/blob/master/packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/scanners/ZxingBarCodeScanner.java#L54

```diff
  public BarCodeScannerResult scan(byte[] data, int width, int height, int rotation) {
    // rotate for zxing if orientation is portrait
    if (rotation == 0) {
-    byte[] rotated = new byte[data.length];
+    byte[] rotated = new byte[Math.max(data.length, width * height)];
      for (int y = 0; y < height; y++) {
        for (int x = 0; x < width; x++) {
          rotated[x * height + height - y - 1] = data[x + y * width];
        }
      }
      width = width + height;
      height = width - height;
      width = width - height;
      data = rotated;
    }

    return scan(generateSourceFromImageData(data, width, height));
  }
```

### Reproducible Demo

I also got similar reports when first releasing my app on the Play store. I got crash reports from Firebase Test Lab automatic tests.

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:354)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=28800; index=30239
	at expo.modules.barcodescanner.scanners.ZxingBarCodeScanner.scan(ZxingBarCodeScanner.java:2)
	at expo.modules.camera.tasks.BarCodeScannerAsyncTask.doInBackground(BarCodeScannerAsyncTask.java:3)
	at expo.modules.camera.tasks.BarCodeScannerAsyncTask.doInBackground(BarCodeScannerAsyncTask.java:1)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
```

--- a/packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/scanners/ZxingBarCodeScanner.java
+++ b/packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/scanners/ZxingBarCodeScanner.java
@@ -54,7 +54,11 @@ public class ZxingBarCodeScanner extends ExpoBarCodeScanner {
       byte[] rotated = new byte[data.length];
       for (int y = 0; y < height; y++) {
         for (int x = 0; x < width; x++) {
-          rotated[x * height + height - y - 1] = data[x + y * width];
+          int sourceIx = x + y * width;
+          int destIx = x * height + height - y - 1;
+          if (sourceIx >= 0 && sourceIx < data.length && destIx >= 0 && destIx < data.length) {
+            rotated[destIx] = data[sourceIx];
+          }
         }
       }
       width = width + height;

